asdf/asdf.py

```diff
@@ -1287,6 +1222,132 @@ def get_history_entries(self):
 AsdfFile.keys.__doc__ = dict.keys.__doc__
 
 
+def _check_and_set_mode(fileobj, asdf_mode):
+
+    if asdf_mode is not None and asdf_mode not in ['r', 'rw']:
```

Comment:
Can we have r+ or w+ modes?

Reply:
Do you just mean that we should add these and make them equivalent to the current `'rw'`? Although I think to be consistent with Python's `open` API, `'w+'` would have to overwrite an existing file. Maybe we should support that, though?